src/finn/util/pyverilator.py

```diff
@@ -189,6 +191,18 @@ def verilator_fifosim(model, n_inputs, max_iters=100000000):
     swg_pkg = os.environ["FINN_ROOT"] + "/finn-rtllib/swg/swg_pkg.sv"
     verilog_file_arg = [swg_pkg, "finn_design_wrapper.v", xpm_memory, xpm_cdc, xpm_fifo]
 
+    additional_verilator_args = []
+    # disable common verilator warnings that should be harmless but commonly occur
+    # in large quantities for Vivado HLS-generated verilog code
+    if disable_common_warnings:
+        additional_verilator_args += ["-Wno-STMTDLY"]
```

Comment:
As discussed, this needs to be integrated for each of the custom-ops... Probably some refactoring should be done to avoid all this code duplication.